reference_test.go

```diff
@@ -41,48 +42,62 @@ func isSubunitInstalled() bool {
-	{"successful-test", "success"},
-	{"unexpected-successful-test", "uxsuccess"},
-	{"skipped-test", "skip"},
-	{"failed-test", "fail"},
-	{"expected-failed-test", "xfail"},
+	{TestID: "existing-test", Status: "exists"},
+	{TestID: "progressing-test", Status: "inprogress"},
+	{TestID: "successful-test", Status: "success"},
+	{TestID: "unexpected-successful-test", Status: "uxsuccess"},
+	{TestID: "skipped-test", Status: "skip"},
+	{TestID: "failed-test", Status: "fail"},
+	{TestID: "expected-failed-test", Status: "xfail"},
 
 	// Different test id lengths.
-	{"test-id (1 byte)", "exists"},
-	{"test-id-with-63-chars (1 byte____)" + strings.Repeat("_", 63-34), "exists"},
-	{"test-id-with-64-chars (2 bytes___)" + strings.Repeat("_", 64-34), "exists"},
-	{"test-id-with-16383-chars (2 bytes)" + strings.Repeat("_", 16383-34), "exists"},
-	{"test-id-with-16384-chars (3 bytes)" + strings.Repeat("_", 16384-34), "exists"},
+	{TestID: "test-id (1 byte)", Status: "exists"},
+	{TestID: "test-id-with-63-chars (1 byte____)" + strings.Repeat("_", 63-34), Status: "exists"},
+	{TestID: "test-id-with-64-chars (2 bytes___)" + strings.Repeat("_", 64-34), Status: "exists"},
+	{TestID: "test-id-with-16383-chars (2 bytes)" + strings.Repeat("_", 16383-34), Status: "exists"},
+	{TestID: "test-id-with-16384-chars (3 bytes)" + strings.Repeat("_", 16384-34), Status: "exists"},
 	// We can't test IDs with more length bytes through the command line.
+
+	// Test with timestamp.
+	// Round to microseconds because python's datetime does not accept nanoseconds.
+	{TestID: "test-with-timestamp", Status: "success", Timestamp: time.Now().Round(time.Microsecond)},
 }
 
 func (s *SubunitReferenceSuite) TestReference(c *check.C) {
-	for _, t := range referencetests {
+	for _, e := range referencetests {
 		var goOutput bytes.Buffer
 		stream := &subunit.StreamResultToBytes{Output: &goOutput}
-		err := stream.Status(t.id, t.status)
-		c.Check(err, check.IsNil, check.Commentf("Error running the go version of subunit"))
+		err := stream.Status(e)
+		c.Check(err, check.IsNil, check.Commentf("Error running the go version of subunit: %s", err))
 
 		cmd := exec.Command("python", "-c", fmt.Sprintf(
 			// FIXME the runnable flag must be a parameter. --elopio - 2015-08-31
-			"import subunit; import sys; subunit.StreamResultToBytes(sys.stdout).status(test_id=%q, test_status=%q, runnable=False)",
-			t.id, t.status))
+			"import datetime; import subunit; import sys; from subunit import iso8601; "+
+				"subunit.StreamResultToBytes(sys.stdout).status(%s)",
+			makePythonArgs(e)))
 		pythonOutput, err := cmd.Output()
-		c.Check(err, check.IsNil, check.Commentf("Error runninng the python version of subunit"))
+		c.Check(err, check.IsNil,
+			check.Commentf("Error runninng the python version of subunit: %s", err))
 
-		c.Check(goOutput.Bytes(), check.DeepEquals, pythonOutput)
+		c.Check(goOutput.Bytes(), check.DeepEquals, pythonOutput,
+			check.Commentf("Wrong stream for event %v", e))
```

Comment:
Sorry, that output included debugging info from my side, the actual output:

fgimenez@arkham:~/workspace/gocode/src/github.com/elopio/subunit$ go test -v ./...
=== RUN   Test

----------------------------------------------------------------------
FAIL: reference_test.go:84: SubunitReferenceSuite.TestReference

reference_test.go:101:
    c.Check(goOutput.Bytes(), check.DeepEquals, pythonOutput,
        check.Commentf("Wrong stream for event %v", e))
... obtained []uint8 = []byte{0xb3, 0x2a, 0x3, 0x24, 0x55, 0xf2, 0xe6, 0x6f, 0xe4, 0x92, 0xea, 0x40, 0x13, 0x74, 0x65, 0x73, 0x74, 0x2d, 0x77, 0x69, 0x74, 0x68, 0x2d, 0x74, 0x69, 0x6d, 0x65, 0x73, 0x74, 0x61, 0x6d, 0x70, 0xda, 0xda, 0x7f, 0x8b}
... expected []uint8 = []byte{0xb3, 0x2a, 0x3, 0x24, 0x55, 0xf3, 0x2, 0x8f, 0xe4, 0x92, 0xea, 0x40, 0x13, 0x74, 0x65, 0x73, 0x74, 0x2d, 0x77, 0x69, 0x74, 0x68, 0x2d, 0x74, 0x69, 0x6d, 0x65, 0x73, 0x74, 0x61, 0x6d, 0x70, 0x21, 0x21, 0x9, 0x17}
... Wrong stream for event {test-with-timestamp success 2015-09-11 16:34:23.613608 +0200 CEST}

OOPS: 11 passed, 1 FAILED
--- FAIL: Test (0.65s)
FAIL
exit status 1
FAIL	github.com/elopio/subunit	0.650s